packages/cli/src/cli/provider/action.provider.tag.ts

```diff
@@ -0,0 +1,93 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import { Aws, LogConfig, TileMetadataTag } from '@basemaps/lambda-shared';
+import {
+    CommandLineAction,
+    CommandLineFlagParameter,
+    CommandLineIntegerParameter,
+    CommandLineStringParameter,
+} from '@rushstack/ts-command-line';
+import { invalidateCache } from '../util';
+import { validateProvider, printProvider } from './provider.util';
+import * as c from 'ansi-colors';
+
+export class ProviderUpdateTagAction extends CommandLineAction {
+    private version: CommandLineIntegerParameter;
+    private tag: CommandLineStringParameter;
+    private commit: CommandLineFlagParameter;
+
+    public constructor() {
+        super({
+            actionName: 'tag',
+            summary: 'Tag a version',
+            documentation: '',
+        });
+    }
+
+    protected onDefineParameters(): void {
+        this.version = this.defineIntegerParameter({
+            argumentName: 'VERSION',
+            parameterLongName: '--version',
+            parameterShortName: '-v',
+            description: 'Version ID',
+            required: false,
+        });
+
+        const validTags = Object.values(TileMetadataTag).filter((f) => f != TileMetadataTag.Head);
+        this.tag = this.defineStringParameter({
+            argumentName: 'TAG',
+            parameterLongName: '--tag',
+            parameterShortName: '-t',
+            description: `tag name  (options: ${validTags.join(', ')})`,
+            required: false,
+        });
+
+        this.commit = this.defineFlagParameter({
+            parameterLongName: '--commit',
+            description: 'Commit to database',
+            required: false,
+        });
+    }
+
+    protected async onExecute(): Promise<void> {
+        const tagInput = this.tag.value!;
+        const version = this.version.value!;
+
+        let tag: TileMetadataTag;
+        switch (tagInput) {
+            case TileMetadataTag.Beta:
+            case TileMetadataTag.Head:
+            case TileMetadataTag.Production:
+                tag = tagInput;
+                break;
+            default:
+                LogConfig.get().fatal({ tag: tagInput }, 'Invalid tag name');
+                console.log(this.renderHelpText());
+                return;
+        }
+
+        const before = await Aws.tileMetadata.Provider.get(tag);
+
+        LogConfig.get().info({ version, tag }, 'Tagging');
+
+        if (this.commit.value) {
+            await Aws.tileMetadata.Provider.tag(tag, version);
+            if (tag === TileMetadataTag.Production) {
```

Comment:
this is a pretty huge invalidation.

also why just prod?

Reply:
Just prod because it is a huge invalidation and we currently only use the production tag to serve wmts files